exchange/websocket/subscriptions.go

```diff
@@ -224,20 +224,34 @@ func (m *Manager) GetSubscriptions() subscription.List {
 // The subscription state is not considered when counting existing subscriptions
 func (m *Manager) checkSubscriptions(conn Connection, subs subscription.List) error {
 	var subscriptionStore *subscription.Store
+	var usedCapacity int
 	if wrapper, ok := m.connections[conn]; ok && conn != nil {
-		subscriptionStore = wrapper.subscriptions
+		if subscriptionStore = wrapper.subscriptions; subscriptionStore == nil {
+			return fmt.Errorf("%w: Websocket.subscriptions", common.ErrNilPointer)
+		}
+		var connSubStore *subscription.Store
+		for _, connSubs := range wrapper.connectionsSubs {
+			if connSubs.Connection == conn {
+				connSubStore = connSubs.Subscriptions
+				break
+			}
+		}
+		if connSubStore == nil {
+			return fmt.Errorf("%w: connection subscription store not found", common.ErrNilPointer)
+		}
+		usedCapacity = connSubStore.Len()
 	} else {
 		subscriptionStore = m.subscriptions
-	}
-	if subscriptionStore == nil {
-		return fmt.Errorf("%w: Websocket.subscriptions", common.ErrNilPointer)
+		if subscriptionStore == nil {
```

Comment:
`subscriptionStore` is checked if it is nil. So, we can move this to the else statement and remove the 	`var subscriptionStore *subscription.Store` declaration. And, use only the `usedCapacity` variable since it is the only variable used below.